### PR TITLE
feat: components api refactoring

### PR DESCRIPTION
**Pull request by pichoemr**
#__ 

## Checklist
- [ ] I didn't over-scope my PR
- [ ] My PR title matches the [commit convention](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] I did not include breaking changes
- [ ] I included unit tests that cover my changes
- [ ] I added/updated the documentation about my changes
- [ ] I made my own code-review before requesting one

## Technical highlight/advice

WAIT FOR https://github.com/lenra-io/components-api/pull/140

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -34,8 +34,10 @@ defmodule ApplicationRunner.MixProject do
     [
       {:credo, "~> 1.6.7", only: [:dev, :test], runtime: false},
       {:dialyxir, "~> 1.2", only: [:dev, :test], runtime: false},
-      {:ex_component_schema,
-       git: "https://github.com/lenra-io/ex_component_schema", ref: "v1.0.0-beta.3"},
+      {
+        :ex_component_schema,
+        git: "https://github.com/lenra-io/ex_component_schema", ref: "21adb5c"
```
Comment (jonas-martinez) [inline mix.exs]: Not up to date

**Review by jonas-martinez — CHANGES_REQUESTED** [review]

File: lib/session/ui_builders/lenra_builder.ex [hunk 2]
```diff
@@ -76,6 +76,7 @@ defmodule ApplicationRunner.Session.UiBuilders.LenraBuilder do
          ui_context,
          view_uid
        ) do
+    # TODO: move with in the case avoid validate schema when unsued
```
Comment (jonas-martinez) [inline lib/session/ui_builders/lenra_builder.ex]: Can you reword this text ? I don't understand it with the current wording.

File: lib/schema/json_shemata.ex [hunk 3]
```diff
@@ -73,19 +86,23 @@ defmodule ApplicationRunner.JsonSchemata do
 
   def get_component_path(comp_type), do: "components/#{comp_type}.schema.json"
 
-  @impl true
-  def handle_call({:load_raw_schema, schema, component_name}, _from, schemata_map) do
-    {:reply, :ok, load_raw_schema(schema, schemata_map, component_name)}
-  end
+  # @impl true
+  # def handle_call({:load_raw_schema, schema, component_name}, _from, schemata_map) do
```
Comment (jonas-martinez) [inline lib/schema/json_shemata.ex]: Do we still need the `load_raw_schema` comments across application_runner ? If not, you can remove every comment that corresponds to the `load_raw_schema`

File: mix.exs (hunk 1 above)
Comment (pichoemr) [reply mix.exs]: We need to wait this PR https://github.com/lenra-io/ex_component_schema merge and switch to beta branch

File: lib/schema/json_shemata.ex [hunk 4]
```diff
@@ -12,57 +12,57 @@ defmodule ApplicationRunner.JsonSchemata do
...
   def init(_) do
-    root_json_directory = Application.app_dir(:application_runner, @component_api_directory)
-
-    relative_shemata_path =
-      Path.join(root_json_directory, "/**/*.schema.json")
-      |> Path.wildcard()
-      |> Enum.map(&Path.relative_to(&1, root_json_directory))
+    root_json_directory =
+      Application.app_dir(:application_runner, @component_api_directory) <>
+        "/component.schema.json"
```
Comment (shiipou) [inline lib/schema/json_shemata.ex]: For easier maintainability, you can define the `"/component.schema.json"` in a const in top of the file near `@component_api_directory`
```suggestion
        @component_api_root_file
```

and in top of the file :
```elixir
@component_api_root_file "/component.schema.json"
```

**Review by shiipou — CHANGES_REQUESTED** [review]

**Review by jonas-martinez — APPROVED** [review]

File: lib/schema/json_shemata.ex [hunk 5]
```diff
@@ -7,62 +7,63 @@ defmodule ApplicationRunner.JsonSchemata do
...
-
-    properties = ApplicationRunner.SchemaParser.parse(resolved_schema)
-
-    [{get_component_path(component_name), Map.merge(%{schema: resolved_schema}, properties)}]
-    |> Enum.into(schemata_map)
+  def load_schema(root_schema) do
+    Map.replace(
+      root_schema,
+      :refs,
+      Map.map(root_schema.refs, fn {id, ref} ->
```
Comment (shiipou) [inline lib/schema/json_shemata.ex:43]: `Map.map` is deprecated use this instead : 

```suggestion
      Map.new(root_schema.refs, fn {id, ref} ->
```

**Review by shiipou — CHANGES_REQUESTED** [review]

**Pull request merged**